### PR TITLE
Added SmartSave

### LINKED ISSUE #453: Cache unsaved text documents
**Is your feature request related to a problem? Please describe.**
Sometimes unexpected bad things might happen and you might lose your work!

**Describe the solution you'd like**
A file cache that store upto `x1` unsaved files upto `x2`MB in size in the app location(?)

**Describe alternatives you've considered**
Caching in ram, but that might be not enough in some scenarios

**Additional context**
Notepad++ implements this feature and it does its job well!

### PR DESCRIPTION
**Pull request by d2dyno1**
This PR adds file caching functionality. Up to `x` files are cached of any size.

Things yet to do:
- [x] Invalidate outdated cache files
- [x] Invalidate cached file when it's saved
- [x] Load cached file
- [x] ~~Show tip if documents are available to be recovered~~
- [x] List of cached files
- [ ] Button to invalidate whole cache

`x` = 10
Closes #453
This PR is a draft.

### CODE REVIEW

**Comment by d2dyno1** [comment]
I've not tested it yet though

**Comment by d2dyno1** [comment]
@yaichenbaum do you have any ideas how would restoring cached files look alike? I need your feedback 👍

**Comment by yaichenbaum** [comment]
@d2dyno1 You can probably add a sub menu to the File menu.

**Comment by d2dyno1** [comment]
@yaichenbaum could you check latest changes?

**Comment by d2dyno1** [comment]
@yaichenbaum one question though: How do I access `MenuBar` from within `WindowsDocumentManager`?

**Comment by yaichenbaum** [comment]
@d2dyno1 What do you need to access in MenuBar?

**Comment by d2dyno1** [comment]
bruh I forgot now. It was like 2 mo ago

File: src/QuickPad.UI/QuickPad.UI/Controls/MenuBar.xaml [hunk 1]
```diff
@@ -73,6 +73,15 @@
                     </MenuFlyoutItem.KeyboardAccelerators>
                 </MenuFlyoutItem>
                 <MenuFlyoutSeparator />
+                <MenuFlyoutSubItem 
+                    x:Uid="CachedFiles"
+                    x:FieldModifier="public"
+                    x:Name="CachedFilesSubMenu"
+                    Text="SmartSave-Restored Documents">
+                    <MenuFlyoutSubItem.Icon>
+                        <FontIcon FontFamily="/Assets/Fonts/fluentui.ttf#fluentui" Glyph="&#xE92D;" />
```
Comment (yaichenbaum) [inline src/QuickPad.UI/QuickPad.UI/Controls/MenuBar.xaml:82]: This font isn't part of Quick Pad anymore, can you sync with `main`?

**Pull request closed without merging**